jlc_kicad_tools/generate_jlc_files.py

```diff
@@ -43,67 +65,62 @@ def main():
 
-	# Default log level is WARNING
-	logging.basicConfig(format="%(message)s", level=max(logging.WARNING - opts.verbose_count * 10, logging.NOTSET))
+def main():
+
+	opts = getOpts()
+
+	_LOGGER.setLevel(opts.verbose_count)
 
 	if not os.path.isdir(opts.project_dir):
-		logging.error("Failed to open project directory: {}".format(opts.project_dir))
+		_LOGGER.logger.error("Failed to open project directory: {}".format(opts.project_dir))
 		return errno.ENOENT
 
 	# Set default output directory
 	if opts.output_dir == None:
 		opts.output_dir = opts.project_dir
 
 	if not os.path.isdir(opts.output_dir):
-		logging.info("Creating output directory {}".format(opts.output_dir))
+		_LOGGER.logger.info("Creating output directory {}".format(opts.output_dir))
 		os.mkdir(opts.output_dir)
 
 	project_name = os.path.basename(opts.project_dir)
-	logging.debug("Project name is '%s'.", project_name)
-	netlist_filename = project_name + ".xml"
-	cpl_filename = project_name + "-all-pos.csv"
-	netlist_path = None
-	cpl_path = None
+	_LOGGER.logger.debug("Project name is '%s'.", project_name)
+
+	netlist = Component(project_name + ".xml")
+	cpl = Component(project_name + "-all-pos.csv")
+	cpl.setErrorMessage("Failed to find CPL file: {} in {} (and sub-directories). "
+	"Run 'File -> Fabrication Outputs -> Footprint Position (.pos) File' in Pcbnew. "
+	"Settings: 'CSV', 'mm', 'single file for board'.")
+	netlist.setErrorMessage("Failed to find netlist file: {} in {} (and sub-directories). "
+	"Is the input directory a KiCad project? "
+	"If so, run 'Tools -> Generate Bill of Materials' in Eeschema (any format). "
+	"It will generate an intermediate file we need. "
+	"Note that this is not the same as a netlist for Pcbnew.")
 
 	for dir_name, subdir_list, file_list in os.walk(opts.project_dir):
 		for file_name in file_list:
-			if file_name == netlist_filename:
-				netlist_path = os.path.join(dir_name, file_name)
-			elif file_name == cpl_filename:
-				cpl_path = os.path.join(dir_name, file_name)
-
-	if netlist_path is None:
-		logging.error((
-			"Failed to find netlist file: {} in {} (and sub-directories). "
-			"Is the input directory a KiCad project? "
-			"If so, run 'Tools -> Generate Bill of Materials' in Eeschema (any format). "
-			"It will generate an intermediate file we need. "
-			"Note that this is not the same as a netlist for Pcbnew.").format(netlist_filename, opts.project_dir))
-		return errno.ENOENT
+			if file_name == netlist.file_name:
+				netlist.setPath(os.path.join(dir_name, file_name))
+			elif file_name == cpl.file_name:
+				cpl.setPath(os.path.join(dir_name, file_name))
 
-	if cpl_path is None:
-		logging.error((
-			"Failed to find CPL file: {} in {} (and sub-directories). "
-			"Run 'File -> Fabrication Outputs -> Footprint Position (.pos) File' in Pcbnew. "
-			"Settings: 'CSV', 'mm', 'single file for board'.").format(cpl_filename, opts.project_dir))
-		return errno.ENOENT
+	netlist.isPathFound(opts.output_dir)
```

Comment:
Got it. Probably I was more laser-focus on trying to create a method itself rather than other aspects. That's why this review process is a core piece to understand what to do bear in mind in future contributions.